### ISSUE TITLE
Fix compiler warnings

### ISSUE THREAD

**Issue by lavinrp** [lavinrp/GameBackbone#186]
MSVC gives a warning for every STL class stored as a member. This can be a real problem for projects that are concerned about ABI, or if we forget to dllexport one of our classes.

GB is not concerned about ABI and we test all of our classes in a separate exe to ensure that everything is exported correctly. This makes the warning of little use to us.

Disable the warning in a way that will not propagate to our clients.
http://cmake.3232098.n2.nabble.com/How-to-disable-specific-warnings-in-VS-IDE-td7589820.html

**Comment by gyurgyma** [comment]
@lavinrp I've fixed all of them in our code as of the current warnings you enables. Just for fun, I turned on W4 to see what it would give us and its quite interesting looking at the warnings:

- Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow (io.2).
This seems really aggressive, but I kinda get it. Here's a link about it: https://stackoverflow.com/questions/55995817/warning-c26451-arithmetic-overflow.

- declaration of 'components' hides class member
This one is properly meaningful. We do this all over the place. I think we should actually look into fixing these, whether with better naming conventions or that super handy 'm_'.

- 'elapsedTime': unreferenced formal parameter
Again, kinda useful, kinda not. So this is mostly complaining about out {} functions. Some people say to fix this by using unnamed variables ei: void foo(int /*name*/){}. I'm not sure about it.
